lib/betamocks/response_cache.rb

```diff
@@ -64,6 +67,10 @@ def file_path(file_name = nil)
       File.join(dir_path, file_name || @generated_file_name)
     end
 
+    def optional_locator_file_path(file_name = nil)
+      File.join(dir_path, OptionalLocator.new(@env).generate, file_name || @generated_file_name)
```

Comment:
Why does this not just return a file name like ```generate_file_name``` does? I'm not totally sure why we are using "generate" on that, just call it file_name and return it. Also, assigning an instance variable in the constructor of a method seems like another opportunity for optimization. This is more related to the whole file, I think there are so many opportunities for refactor here that I could pick at it for a whole day.

1. Remove generate file_name from initializer and just call the method when needed.
```
def initialize
  @env = env
  @config = config
end
```

2. Convert attr_writer to attr_accessor and don't use the instance variables all over. 

3. Return the same thing for optional_locator and file_name. 
```
def optional_locator_file_name
  OptionalLocator.new(@env).generate, file_name || file_name)
end

def file_name
  name = @config[:file_path].split('/').last
  @config.dig(:cache_multiple_responses) ? "#{Uid.new(@env).generate}.yml" : "#{name}.yml"
end
```